jazzhands/__main__.py

```diff
@@ -184,25 +209,57 @@ def main(argv=sys.argv):
     # collect, build, run or the setup command.
     # build implies collect, run implies build and collect. setup runs on its own.
 
-    if len(sys.argv) <= 1:
+    parser = argparse.ArgumentParser()
+    parser.set_defaults(which=None)
+    # parser.add_argument('command', type=str, action='store')
+    subparsers = parser.add_subparsers()
+    parsers = {}
+    def add_command(cmd):
+        parsers[cmd] = subparsers.add_parser(cmd)
+        parsers[cmd].set_defaults(which=cmd)
+
+    add_command('setup')
+    parsers['setup'].add_argument('-p', '--preset', action='append', type=str, default=[])
+    parsers['setup'].add_argument('-t', '--transform', action='append', type=str, default=[])
```

Comment:
I never knew you could do this with argparse.

Reply:
Neither did I!